buckets/buckets.go

```diff
@@ -212,7 +212,24 @@ func (b *Buckets) Save(ctx context.Context, dbID thread.ID, bucket *Bucket, opts
 	for _, opt := range opts {
 		opt(args)
 	}
-	return b.threads.Save(ctx, dbID, CollectionName, dbc.Instances{bucket}, db.WithTxnToken(args.Token))
+	ensureNoNulls(bucket)
+	err := b.threads.Save(ctx, dbID, CollectionName, dbc.Instances{bucket}, db.WithTxnToken(args.Token))
+	if err != nil {
+		if isInvalidSchemaErr(err) {
+			if err := b.updateCollection(ctx, dbID, opts...); err != nil {
+				return err
+			}
+			return b.Save(ctx, dbID, bucket, opts...)
+		}
+		return err
+	}
+	return nil
+}
+
+func ensureNoNulls(b *Bucket) {
```

Comment:
We'll have to continue to stack field migrations here until there's a better way (see https://github.com/textileio/go-threads/issues/363)